community/kernel/src/main/java/org/neo4j/kernel/impl/core/GraphPropertiesProxy.java

```diff
@@ -195,6 +198,45 @@ public Object removeProperty( String key )
     }
 
     @Override
+    public Map<String, Object> getProperties( String... names )
+    {
+        try ( Statement statement = actions.statement() )
+        {
+            Map<String, Object> properties = new HashMap<>();
+            ReadOperations readOperations = statement.readOperations();
+            for ( String name : names )
+            {
+                int propertyKeyId = readOperations.propertyKeyGetForName( name );
+                properties.put( name, readOperations.graphGetProperty( propertyKeyId ) );
```

Comment:
That might actually be desirable (since it means the set of keys is maintained), I wonder if we should make the other implementations do that as well, or if we should change this implementation, they should at least have the same behaviour in that regard (although this implementation is of course never used by end-user code).

This is something we should discuss IMO.